SurveyAppTests/Tests/Modules/Login/LoginPresenterSpec.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  LoginPresenterSpec.swift
+//  SurveyApp Tests
+//
+//  Created by Chananchida F. on 6/17/21.
+//
+
+import Quick
+import Nimble
+
+@testable import SurveyApp
+
+final class LoginPresenterSpec: QuickSpec {
+
+    override func spec() {
+        var router: LoginRouterInputMock!
+        var interactor: LoginInteractorInputMock!
+        var view: LoginViewInputMock!
+        var output: LoginOutputMock!
+        var presenter: LoginPresenter!
+
+        describe("a LoginPresenter") {
+            beforeEach {
+                router = LoginRouterInputMock()
+                interactor = LoginInteractorInputMock()
+                output = LoginOutputMock()
+                view = LoginViewInputMock()
+                presenter = LoginPresenter(router: router, interactor: interactor)
+
+                presenter.output = output
+                presenter.view = view
+
+            }
+
+            context("when didPressLogin is called with valid email and password") {
+                beforeEach {
+                    let email = "email@example.com"
+                    let password = "password"
+                    presenter.didPressLogin(email: email, password: password)
+                }
+
+                it("triggers interactor authenticateEmail") {
+                    expect(interactor.authenticateWithEmailEmailPasswordCalled) == true
+                }
+            }
+
+            context("when didPressLogin is called with empty email and password") {
+                beforeEach {
+                    let email = ""
+                    let password = ""
+                    presenter.didPressLogin(email: email, password: password)
+                }
+
+                it("should not triggers interactor authenticateEmail") {
+                    expect(interactor.authenticateWithEmailEmailPasswordCalled) == false
+                }
+            }
+
+            context("when didPressLogin is called with invalid email") {
+                beforeEach {
+                    let email = "email"
+                    let password = "password"
+                    presenter.didPressLogin(email: email, password: password)
+                }
+
+                it("triggers view to show error") {
```

Comment:
You may want to unify about this pattern for the others
```
triggers view to show error
should not trigger interactor to authenticate with email
or
triggers view to call showErrorMessage
should not trigger interactor to call authenticateEmail
```

Reply:
Thank you. I'll make that change 🙇‍♀️